robotoff/app/api.py

```diff
@@ -776,19 +777,29 @@ def on_post(self, req: falcon.Request, resp: falcon.Response):
             logo_id = annotation["logo_id"]
             type_ = annotation["type"]
             value = annotation["value"] or None
-            logo = LogoAnnotation.get_by_id(logo_id)
+            try:
+                logo = LogoAnnotation.get_by_id(logo_id)
+            except LogoAnnotation.DoesNotExist:
+                raise falcon.HTTPNotFound(description=f"logo {logo_id} not found")
+
             if value is not None:
                 logo.annotation_value = value
                 value_tag = get_tag(value)
                 logo.annotation_value_tag = value_tag
                 logo.taxonomy_value = match_taxonomized_value(value_tag, type_)
+            elif type_ in ("brand", "category", "label", "store"):
+                raise falcon.HTTPBadRequest(
+                    description=f"value required for type {type_} (logo {logo_id})"
+                )
 
             logo.annotation_type = type_
             logo.username = username
             logo.completed_at = completed_at
-            logo.save()
             annotated_logos.append(logo)
 
+        for logo in annotated_logos:
+            logo.save()
```

Comment:
Why doing another loop ?

Reply:
So that we check that every logo is correct before saving them in DB.
Previously we could save some of the and throw an HTTP 400 error.